File: app/src/main/java/denis/easyweather/app/inject/modules/MainActivityModule.java
package denis.easyweather.app.inject.modules;

import dagger.Module;
import dagger.Provides;
import denis.easyweather.app.inject.scopes.ActivityScope;
import denis.easyweather.app.router.MainActivity;

@Module
public class MainActivityModule {
    private final MainActivity mainActivity;

    public MainActivityModule(MainActivity mainActivity) {
        this.mainActivity = mainActivity;
    }

    @Provides
    @ActivityScope
    public MainActivity mainActivity() {
        return mainActivity;
    }
}

File: app/src/main/java/denis/easyweather/app/inject/modules/MainActivityModule.kt
package denis.easyweather.app.inject.modules

import dagger.Module
import dagger.Provides
import denis.easyweather.app.inject.scopes.ActivityScope
import denis.easyweather.app.router.MainActivity

@Module
class MainActivityModule(private val mainActivity: MainActivity) {

    @Provides
    @ActivityScope
    fun mainActivity(): MainActivity {
        return mainActivity
    }
}
